### PR TITLE
WIP: react-router 4, graphql

### PR DESCRIPTION
**Pull request by mmahalwy**
Fixes https://github.com/quran/quran.com-frontend/issues/675

Might fix https://github.com/quran/quran.com-frontend/issues/687
Might fix https://github.com/quran/quran.com-frontend/issues/830

Todos:
- [ ] Audio segments
- [ ] Check and validate surah range

### CODE REVIEW

**Comment by mmahalwy** [comment]
oh gosh

**Comment by mmahalwy** [comment]
@ahmedre I think this will cut our scripts size by alot.

File: package.json [hunk 1]
```diff
@@ -104,7 +106,7 @@
     "promise": "7.1.1",
     "proxy-middleware": "0.14.0",
     "qs": "6.2.1",
-    "quran-components": "^0.0.73",
+    "quran-components": "^0.0.79",
```
Comment (sabeurthabti) [inline package.json]: 🔥🔥🔥🔥

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32918

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32921

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32941

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32951

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32952

**Pull request closed without merging**